lib/engine/game/g_18_co.rb

```diff
@@ -90,6 +109,35 @@ def init_stock_market
         )
       end
 
+      def mines_count(entity)
+        entity.abilities(:mine_income).sum(&:count_per_or)
+      end
+
+      def mine_multiplier(entity)
+        entity.abilities(:mine_multiplier).collect(&:count_per_or).reject(&:zero?).inject(1, :*)
```

Comment:
Throughout the codebase, `map` is used instead of `collect`, and `reduce` is used instead of `inject`.